neo4j/driver_with_context.go

```diff
@@ -382,66 +352,187 @@ func (d *driverWithContext) Close(ctx context.Context) error {
+//
+//		neo4j.ExecuteQueryWithDatabase
+//		neo4j.ExecuteQueryWithWritersRouting
+//	 ...
+//
+// see neo4j.ExecuteQueryConfiguration for all possibilities.
+//
+// These built-in callbacks can be used and combined as follows:
+//
+//	ExecuteQuery[T](ctx, driver, query, params, transformerFunc, neo4j.ExecuteQueryWithDatabase("my-db"), neo4j.ExecuteQueryWithWritersRouting())
+//
+// For complete control over the configuration, you can also define your own callback:
+//
+//	ExecuteQuery[T](ctx, driver, query, params, transformerFunc, func(config *neo4j.ExecuteQueryConfiguration) {
+//		config.Database = "my-db"
+//		config.RoutingControl = neo4j.Writers
+//	})
+//
+// ExecuteQuery causal consistency is guaranteed by default across different successful calls to ExecuteQuery
+// targeting the same database.
+// In other words, a successful read query run by ExecuteQuery is guaranteed to be able to read results created
+// from a previous successful write query run by ExecuteQuery on the same database.
+// This is achieved through the use of a bookmarks, managed by a default neo4j.BookmarkManager instance.
+// This default BookmarkManager instance can be retrieved with DriverWithContext.DefaultExecuteQueryBookmarkManager.
+// Such a consistency guarantee is *not* maintained between ExecuteQuery calls and the lower-level
+// neo4j.SessionWithContext API calls, unless sessions are explicitly configured with the same bookmark manager.
+// That guarantee may also break if a custom implementation of neo4j.BookmarkManager is provided via for instance
+// the built-in callback neo4j.ExecuteQueryWithBookmarkManager.
+// You can disable bookmark management by passing the neo4j.ExecuteQueryWithoutBookmarkManager callback to ExecuteQuery.
+//
+// The equivalent functionality of ExecuteQuery can be replicated with pre-existing APIs as follows:
+//
+//	 // all the error handling bits have been omitted for brevity (do not do this in production!)
+//		session := driver.NewSession(ctx, neo4j.SessionConfig{
+//			DatabaseName:     "<DATABASE>",
+//			ImpersonatedUser: "<USER>",
+//			BookmarkManager:  bookmarkManager,
+//		})
+//		defer handleClose(ctx, session)
+//		// session.ExecuteRead is called if the routing is set to neo4j.Readers
+//		result, _ := session.ExecuteWrite(ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+//			result, _ := tx.Run(ctx, "<CYPHER>", parameters)
+//			records, _ := result.Collect(ctx) // real implementation does not use Collect
+//			keys, _ := result.Keys()
+//			summary, _ := result.Consume(ctx)
+//			return &neo4j.EagerResult{
+//				Keys:    keys,
+//				Records: records,
+//				Summary: summary,
+//			}, nil
+//		})
+//		eagerResult := result.(*neo4j.EagerResult)
+//		// do something with eagerResult
+//
+// The available ResultTransformer implementation, EagerResultTransformer, computes an *EagerResult.
+// As the latter's name suggests, this is not optimal when the result is made of an important number of records.
```

Comment:
should this be 'made from a large number or records'?

Reply:
thanks fixed in https://github.com/neo4j/neo4j-go-driver/pull/435/commits/016a0d2ee5b4fd4d30a291d489d8680fe256f53c